src/obs_inv_utils/discover_interface.py

```diff
@@ -0,0 +1,276 @@
+# 2024-04-16
+# discover_interface.py
+# Seth Cohen
+
+import os
+import subprocess
+import re
+from pathlib import Path
+from collections import namedtuple, OrderedDict
+import attr
+from datetime import datetime
+
+DiscoverCommand = namedtuple(
+    'DiscoverCommand',
+    [
+        'command',
+        'arg_validator',
+        'output_parser'
+    ],
+)
+
+DiscoverCommandRawResponse = namedtuple(
+    'DiscoverCommandRawResponse',
+    [
+        'command',
+        'return_code',
+        'error',
+        'output',
+        'success',
+        'args_0',
+        'submitted_at',
+        'latency'
+    ],
+)
+
+DiscoverListContents = namedtuple(
+    'DiscoverListContents',
+    [
+        #'parent_dir',
+        'prefix',
+        'files_count',
+        'files_meta',
+        'obs_cycle_time',
+        'submitted_at',
+        'latency'
+    ],
+)
+
+HpssTarballContents = namedtuple(
+    'HpssParsedTarballContents',
+    [
+        'parent_dir',
+        'expected_count',
+        'inspected_files',
+        'observation_day',
+        'submitted_at',
+        'latency'
+    ],
+)
+
+
+
+HpssFileMeta = namedtuple(
+    'HpssFileMeta',
+    [
+        'name',
+        'permissions',
+        'last_modified',
+        'size'
+    ],
+)
+
+DiscoverFileMeta = namedtuple(
+    'DiscoverFileMeta',
+    [
+        'name',
+        'permissions',
+        'last_modified',
+        'size',
+        'etag'
+    ],
+)
+
+
+CMD_GET_DISCOVER_OBJ_LIST = 'list_discover'
+EXPECTED_COMPONENTS_DISCOVER_OBJ_LIST = 8
+
+
+# A ~ 'command'
+#['ls', '-l','--time-style=+"%Y-%m-%d %H:%M:%S "']
+
+# B ~ 'arg_validator'
+def inspect_discover_args_valid(arg):
```

Comment:
Is this just a placeholder function for later validation logic?

Reply:
Correct. This was left as a placeholder as I worked out the main functionality (the 'arg_parser' in part C). I need to revisit this. Thank you for bringing to my attention.